src/lib/Service/UserPreferences/DateTimeFormat/DateTimeFormatRepository.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\EzPlatformAdminUi\Service\UserPreferences\DateTimeFormat;
+
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+
+class DateTimeFormatRepository
+{
+    /**
+     * @var array
+     */
+    private $allowedDateFormats;
+
+    /**
+     * @var array
+     */
+    private $allowedTimeFormats;
+
+    /**
+     * @var \EzSystems\EzPlatformAdminUi\Service\UserPreferences\DateTimeFormat\DateTimeFormatTransformer
+     */
+    private $transformaer;
+
+    /**
+     * @param \eZ\Publish\Core\MVC\ConfigResolverInterface $configResolver
+     */
+    public function __construct(ConfigResolverInterface $configResolver)
+    {
+        $this->allowedDateFormats = $configResolver->getParameter('user_preferences.allowed_date_formats');
```

Comment:
This can be injected directly from the service definition: `$user_preferences.allowed_date_formats$`